### ISSUE TITLE
2.0 provider compatibility: HistoryRepository generates INSERT statements without quotes

### ISSUE THREAD

**Issue by divega** [dotnet/efcore#11696]
(This was reported via email)

Creating an issue although there is a version mismatch because this could be a symptom that other 2.0 providers that update less often than Npgsql could be broken as well:

Microsoft.EntityFrameworkCore.PostgreSQL version is 2.0.1 
Microsoft.EntityFrameworkCore version is 2.1.0-preview2

The INSERT statement into the __EFMigrationsHistory is missing single quotes for the VALUES listed:

Expected (working SQL from 2.0.2): 
``` SQL 
INSERT INTO "_EFMigrationsHistory" ("MigrationId", "ProductVersion") VALUES ('20180413232313Init2', '2.0.2-rtm-10011')
```

Actual (quotes missing, invalid SQL):
``` SQL
INSERT INTO "__EFMigrationsHistory" ("MigrationId", "ProductVersion") VALUES (20180413233035_Init3, 2.1.0-preview2-30571) &lt;
```

**Comment by bricelam** [comment]
@ajcvickers This looks like an issue with the `FallbackRelationalTypeMappingSource`.
``` csharp
class Context : DbContext
{
    protected override void OnConfiguring(DbContextOptionsBuilder options)
        => options.UseNpgsql("Data Source=");
}

static void Main()
{
    var db = new Context();

    var typeMapper = db.GetService<IRelationalTypeMapper>();
    var oldTypeMapping = typeMapper.FindMapping(typeof(string));
    var oldLiteral = oldTypeMapping.GenerateSqlLiteral("test");
    Console.WriteLine($"Before:\t{oldLiteral}\t({oldTypeMapping.GetType().Name})");

    var typeMappingSource = db.GetService<IRelationalTypeMappingSource>();
    var newTypeMapping = typeMappingSource.FindMapping(typeof(string));
    var newLiteral = newTypeMapping.GenerateSqlLiteral("test");
    Console.WriteLine($"After:\t{newLiteral}\t({newTypeMapping.GetType().Name})");
}
```
Output:
```
Before: 'test'  (NpgsqlStringTypeMapping)
After:  test    (NpgsqlTypeMapping)
```

**Comment by bricelam** [comment]
P.S. As far as I can tell, only Npgsql is affected for the Migrations path.

**Comment by roji** [comment]
Thanks for looking into this and sorry for the bother, it's a duplicate of https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/issues/283 which has already been fixed for 2.0.2 (not yet released).

**Comment by roji** [comment]
FYI released 2.0.2 yesterday so this should be ok now.
